lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+        set_background_color(color::Lcha),
+
+        /// Set the slider value
+        set_value(f32),
+        /// Set the default value to reset a slider to
+        set_value_default(f32),
+        /// Set the minimum to which the value is clamped
+        set_value_min(f32),
+        /// Set the maximum to which the value is clamped
+        set_value_max(f32),
+        /// Set the color of the value text display
+        set_value_color(color::Lcha),
+
+        /// Set the default precision at which the slider operates
+        set_precision(f32),
+        /// Set the margin above/below the slider beyond which the precision is adjusted up/downwards
+        set_precision_step_margin(f32),
+        /// Set the distance of vertical mouse movement needed to increment/decrement the precision to the next step
+        set_precision_step_size(f32),
+
+        /// Set a slider label
+        set_label(ImString),
+        /// Set the color of the slider label
+        set_label_color(color::Lcha),
+        /// Set whether the label is displayed
+        set_label_visible(bool),
+        /// Set whether the label is shown inside the slider, as opposed to left of it
+        set_label_inside(bool),
+
+        /// Set whether the slider is enabled, when disabled the slider is greyed out and cannot be interacted with
+        set_slider_enabled(bool),
+    }
+    Output {
+        width(f32),
+        height(f32),
+        value(f32),
+    }
+}
+
+impl Slider {
+    /// Initialise a slider component
+    fn init(self) -> Self {
+        let network = self.frp.network();
+        let input = &self.frp.input;
+        let output = &self.frp.private.output;
+        let model = &self.model;
+
+        let scene = &self.app.display.default_scene;
+        let mouse = &scene.mouse.frp;
+        let keyboard = &scene.keyboard.frp;
+
+        let track_pos = Animation::new_non_init(network);
+
+        let background_color = color::Animation::new(network);
+        let track_color = color::Animation::new(network);
+        let value_color = color::Animation::new(network);
+        let label_color = color::Animation::new(network);
+
+        frp::extend! { network
+
+            // User input
+
+            background_click        <- model.background.events.mouse_down_primary.constant(());
+            background_release      <- model.background.events.mouse_release_primary.constant(());
+            background_drag         <- bool(&background_release, &background_click);
+
+            track_click             <- model.track.events.mouse_down_primary.constant(());
+            track_release           <- model.track.events.mouse_release_primary.constant(());
+            track_drag              <- bool(&track_release, &track_click);
+
+            component_click         <- any2(&background_click, &track_click).gate(&input.set_slider_enabled);
+            component_drag          <- any2(&background_drag, &track_drag).gate(&input.set_slider_enabled);
```

Comment:
In old code we were doing that, sorry if you have seen it. In new code, we are not aligning vertically `=` nor `<-`. This alignment is from times when we were not suing auto-formatter.